lib/mri/uri/generic.rb

```diff
@@ -591,10 +591,7 @@ def check_host(v)
       if @opaque
         raise InvalidURIError,
           "can not set host with registry or opaque"
-      elsif !parser.regexp[:HOST].match(v)
-        # Truffle: !~ would cause a deopt to set named match
-        # variables. Since we are only interested in whether the
-        # method matches we should avoid this cost.
+      elsif parser.regexp[:HOST] !~ v
```

Comment:
@aardvark179 Is that workaround still needed?

Reply:
Will restore for now anyway.